### PR TITLE
Todolist

### PR DESCRIPTION
**Pull request by yosefHesham**
**Description**

Added a bunch of code to create a full todo list web app

---

**Changes Made:**

- Create the UI with dynamic js
- Created ToDo item class.
- Created the ToDo controller class which is responsible for the business logic.
- Created CRUD Operations inside the TODO controller class.
- Created module for JSON parsing.
- Created module for local storage.
- Created module for marking the task as completed

### CODE REVIEW

**Review by bizip — APPROVED** [review]
Hi @yosefHesham 


Great work on making your project up to this point 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on as optional changes, but you deserve approval there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️
- Functionality. ✔️


## Optional suggestions ♻️
_Check the comments under the review._



- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me `@bizip` in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/change_status.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        ToDoController.changeStatus(Number(event.target.classList[1]));
+      });
+    }
+  });
+};
+
+const refreshCheckBoxListeners = () => {
+  const checkBoxes = document.querySelectorAll('.item-check');
+  checkBoxes.forEach((checkBox) => {
+    checkBox.addEventListener('click', (event) => {
```
[the remark this answers is not in the document]
Comment (yosefHesham) [reply modules/change_status.js:24]: Which function? @bizip

File: src/style.css [hunk 2]
```diff
@@ -0,0 +1,80 @@
+
```
[the remark this answers is not in the document]
Comment (yosefHesham) [reply src/style.css:1]: You are right

File: modules/change_status.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bizip) [reply modules/change_status.js:24]: I mean to refactor this one:

image

I am expecting something like:

`checkBox.addEventListener('click',callYourFunction);`

**But it is optional**

**Pull request merged**